client/my-sites/themes/controller.js

```diff
@@ -49,27 +45,43 @@ function getProps( context ) {
 	};
 
 	return {
-		title,
 		tier,
 		search: context.query.s,
 		trackScrollPage: boundTrackScrollPage,
 		runClientAnalytics
 	};
-}
+};
 
-function makeElement( ThemesComponent, Head, store, props ) {
-	return(
-		<ReduxProvider store={ store }>
-			<Head title={ props.title } tier={ props.tier || 'all' }>
-				<ThemesComponent { ...omit( props, [ 'title', 'runClientAnalytics' ] ) } />
+const LoggedInHead = ( { children, context: { params: { tier, site_id: siteID } } } ) => (
+	<Head
+		title={ buildTitle(
+			i18n.translate( 'Themes', { textOnly: true } ),
+			{ siteID }
+		) }
+		tier={ tier || 'all' }>
+		{ children }
+	</Head>
+);
+
+// This is generic -- nothing themes-specific in here!
+function makeElement( Component, getProps, Head, sideEffects = function() {} ) {
+	// How do we handle the dispatch?
+	return ( context, next ) => {
```

Comment:
> Could we cache by ( route, context ) instead?

… Mmmmaybe? :)

I assumed from the start that all of `context` would be something unpractical to apply equality checks to. Plus, we'll likely never want to compare `store` inside it. So I guess we could extract a subset of `context` and use that as part of the caching key? Or rather use `omit` (so that devs can attach additional properties to `context` and have that count for caching).